anyway/flask_app.py

```diff
@@ -1461,6 +1461,41 @@ def load_user(id: str) -> Users:
     return db.session.query(Users).get(id)
 
 
+# TODO: in the future add pagination if needed
+@app.route("/admin/get_all_users_info")
+def get_all_users_info() -> Response:
+    if current_user.is_anonymous:
+        return return_json_error(Es.BR_USER_NOT_LOGGED_IN)
+
+    admin_permission = Permission(RoleNeed(BE_CONST.Roles2Names.Admins.value))
+    if not admin_permission.can():
+        return return_json_error(Es.BR_MISSING_PERMISSION, BE_CONST.Roles2Names.Admins.value)
```

Comment:
Where is it configured that `admin` role is required to get users info? Is it hard-coded here?

Reply:
The answer for all of your questions is in line 1471, this checks if the current login user has the admin role to get users info.

In more details:
There is a global object that represent the current user roles, this object is being created and maintained by the `flask-principal` package, when calling the `admin_permission.can()` function the package check if the object that represent the current user has the admin role.

Reply:
Hi @BarVolunteering , I still miss something, where is it configured what roles have permission for this specific action (i.e. list users info)?

Reply:
Not sure that i understand, I send you a message on slack.